contracts/cw1-whitelist-ng/src/interfaces.rs

```diff
@@ -0,0 +1,40 @@
+use crate::msg::AdminListResponse;
+use cosmwasm_std::{CosmosMsg, Deps, DepsMut, Env, MessageInfo, Response};
+use cw1::query::CanExecuteResponse;
+
+pub trait Cw1Whitelist<T> {
+    type Error;
+
+    fn execute(
+        &self,
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        msgs: Vec<CosmosMsg<T>>,
+    ) -> Result<Response<T>, Self::Error>;
+
+    fn freeze(
+        &self,
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+    ) -> Result<Response<T>, Self::Error>;
+
+    fn update_admins(
+        &self,
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        admins: Vec<String>,
+    ) -> Result<Response<T>, Self::Error>;
+
+    fn admin_list(&self, deps: Deps, env: Env) -> Result<AdminListResponse, Self::Error>;
```

Comment:
I completely don't see a reason why `Whitelist` would have any relation to `Cw1`. I can have whitelist restricting any other contract, those are two separate ideas.

Reply:
Fair enough. No need to have the traits extend each other

Let's get it to work now as one trait.
Having it as two separate traits and showing how the messages work with that would be a great demo to show composition. This can be a future step, just see how you can demo that even in one contract.